src/nanorc/sshpm.py

```diff
@@ -154,6 +155,85 @@ def execute_script(self, script_data):
             proc = sh.ssh(ssh_args)
             self.log.info(proc)
 
+    def setup_app(self, app_name, app_conf, conf_loc):
+        hosts = self.boot_info["hosts-ctrl"]
+        rte_script = self.boot_info.get('rte_script')
+        env_vars = self.boot_info["env"]
+
+        host = hosts[app_conf["host"]]
+
+        env_formatter = {
+            "APP_HOST": host,
+            "DUNEDAQ_PARTITION": env_vars['DUNEDAQ_PARTITION'],
+            "APP_NAME": app_name,
+            "APP_PORT": app_conf["port"],
+            "APP_WD": os.getcwd(),
+            "CONF_LOC": conf_loc,
+        }
+
+        if 'update-env' in app_conf:
```

Comment:
Another comment. Why do we need `update-env` here? Can't the base port for the DAQ applications and service be treated in the same way? I see that the port is already offset in line 219 of `cfgmgr.py`.

Reply:
OK, I think I got it now, this is updating the DAQ Application environment variable

Reply:
This is updating an environment variable (specifically in case of the connectivity service CONNECTION_PORT) with the updated port number. This is not necessary for the applications as they get given their updated port numbers directly in their configuration.